src/main/java/com/amazon/opendistroforelasticsearch/ad/model/DetectorValidationIssue.java

```diff
@@ -0,0 +1,144 @@
+ * and its type is FEATURE_ATTRIBUTES, because it is related to feature;
+ * message would be the message from thrown exception;
+ * subIssues are issues for each feature;
+ * suggestion is how to fix the issue/subIssues found
+ */
+public class DetectorValidationIssue implements ToXContentObject, Writeable {
+    private static final String MESSAGE_FIELD = "message";
+    private static final String SUGGESTED_FIELD_NAME = "suggested_value";
+
+    private final ValidationAspect aspect;
+    private final DetectorValidationIssueType type;
+    private final String message;
+    private Map<String, String> subIssues;
+    private Object suggestion;
+
+    public ValidationAspect getAspect() {
+        return aspect;
+    }
+
+    public DetectorValidationIssueType getType() {
+        return type;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public Map<String, String> getSubIssues() {
+        return subIssues;
+    }
+
+    public Object getSuggestion() {
+        return suggestion;
+    }
+
+    public DetectorValidationIssue(
+        ValidationAspect aspect,
+        DetectorValidationIssueType type,
+        String message,
+        Map<String, String> subIssues,
+        Object suggestion
+    ) {
+        this.aspect = aspect;
+        this.type = type;
+        this.message = message;
+        this.subIssues = subIssues;
+        this.suggestion = suggestion;
+    }
+
+    public DetectorValidationIssue(StreamInput input) throws IOException {
+        aspect = input.readEnum(ValidationAspect.class);
+        type = input.readEnum(DetectorValidationIssueType.class);
+        message = input.readString();
+        if (input.readBoolean()) {
+            subIssues = input.readMap(StreamInput::readString, StreamInput::readString);
+        }
+        if (input.readBoolean()) {
+            suggestion = input.readGenericValue();
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeEnum(aspect);
+        out.writeEnum(type);
+        out.writeString(message);
+        if (subIssues != null && !subIssues.isEmpty()) {
+            out.writeBoolean(true);
+            out.writeMap(subIssues, StreamOutput::writeString, StreamOutput::writeString);
+        } else {
+            out.writeBoolean(false);
+        }
+        if (suggestion != null) {
+            out.writeBoolean(true);
+            out.writeGenericValue(suggestion);
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject().startObject(type.getName());
+        xContentBuilder.field(MESSAGE_FIELD, message);
+        if (subIssues != null) {
+            for (Map.Entry<String, String> entry : subIssues.entrySet()) {
+                xContentBuilder.field(entry.getKey(), entry.getValue());
```

Comment:
Why we expand all sub issues here? How about we put all sub issues in `sub_issues` field ?

Reply:
sure. will add `sub_issues` field.